qa/rpc-tests/test_framework/mininode.py

```diff
@@ -591,7 +716,21 @@ def deserialize(self, f):
         self.hashMerkleRoot = deser_uint256(f)
         self.nTime = struct.unpack("<I", f.read(4))[0]
         self.nBits = struct.unpack("<I", f.read(4))[0]
+        extended = False
+        if self.nBits & 0x00800000:
```

Comment:
This may become obvious later, but how come nBits is guaranteed to not use the extended bit? Does it AND-out it during retargeting? Wouldn't it be better to use the lowest (least impact) bit? I'll delete this comment if I run into the answer before you tell me.

Reply:
The bit is only used in the serialization format, so it is masked out on read and doesn't affect the runtime. As to why that bit is unused: it's the sign bit. The proof-of-work target is never a negative number.

Reply:
Understood! I would expect the sign bit to be `0x80000000` (this is the compact arith_uint256 representation yeah? I didn't even know those could be negative).

Reply:
This is the `SetCompact()` representation, yes, which is inherited from the old openssl bignum representation. OpenSSL allowed negative numbers, so bitcoin's serialization accommodates that as well.  That's why a diff-1 block has `nBits = 0x1d00ffff`; the zero byte in the middle is because `nBits = 0x1cffffff` would be a negative number.

So the representation encodes signed numbers, even though a negative number is disallowed at a higher semantic layer from the way `nBits` is used. So that bit must always be clear in any block header, and is useful for indicating the presence of extended serialization.